### DISCUSSION TITLE
How to update editor content in a child TipTap component based prop change from calling parent

### DISCUSSION THREAD

**Discussion by alp4031** [ueberdosis/tiptap#2312] [General Questions]
The following works great for initializing a TipTap editor in a child component that is called from App.js. I can't figure out how to change the content of the editor after the first rendering when there is a change in the editorContent prop from App.js

In App.js I set editorContent with
```js
const [editorContent, setEditorContent] = useState(`<h2>Hi there,</h2><p>This is a <em>basic</em> example of <strong>tiptap</strong>. Sure, there are all kind of basic text styles you’d probably expect from a text editor. But wait until you see the lists:</p>`);
```

From App.js I call the TipTap child component as follows: 
```js
<TipTapDefault  editorContent={editorContent} />
```

The TipTapDefault component begins with 
```js
export default function TipTapDefault ({editorContent}) {
 const editor = useEditor({
    editable: true,
       extensions: [
      StarterKit,
      TextAlign.configure({
        types: ['heading', 'paragraph'],
      }),
      Highlight,
    ],
    content: editorContent,
  })
```
Using content: editorContent works great and displays the App.js editorContent passed as a prop to the editor as expected. 

When I change the editorContent variable in App.js and setEditorContent, the change is not reflected in the TipTapDefault component's content. I do know that useState is asynchronous. 

How can I get a change in the App.js editorContent variable to be reflected in the TipTapDefault editor's content. I appreciate any help in pointing me in the right direction to figure this out.

**Comment by alp4031** [comment]
My workaround for this until I find a better solution is to create the TipTap editor in the parent, App.js. Then when a new learning activity is loaded into App.js, the editor is updated with editor.commands.setContent(loadedEditorContent); The loadedEditorContent was originally extracted by the author of the activity using editor.getHTML(). Then it is copied and pasted into a text file. This allows learning activities to be distributed within underserved communities without the cost of maintaining a database.

**Reply by rubenesque-code** [reply]
A solution for React is to add the key prop to the TipTap component and change the key to force a rerender. It's not ideal for me tough and it would be good to be able use TipTap as with a React controlled input.
